src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ClimbingPick.java

```diff
@@ -0,0 +1,147 @@
+public class ClimbingPick extends SimpleSlimefunItem<ItemUseHandler> implements DamageableItem {
+
+    private final Map<String, Double> materialSpeedsDef = new HashMap<>();
+    private final ItemSetting<Map<String, Double>> materialSpeeds = new ItemSetting<>("launch-amounts", materialSpeedsDef);
+    private final Set<UUID> users = new HashSet<>();
+
+    public ClimbingPick(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe);
+
+        for (Material mat : MaterialCollections.getAllIceBlocks()) {
+            materialSpeedsDef.put(mat.name(), 1D);
+        }
+        for (Material mat : MaterialCollections.getAllConcretePowderColors()) {
+            materialSpeedsDef.put(mat.name(), 1D);
+        }
+        for (Material mat : MaterialCollections.getAllTerracottaColors()) {
+            materialSpeedsDef.put(mat.name(), 1D);
+        }
+
+        materialSpeedsDef.put(Material.GRAVEL.name(), 0.4);
+        materialSpeedsDef.put(Material.SAND.name(), 0.4);
+        materialSpeedsDef.put(Material.STONE.name(), 0.6D);
+        materialSpeedsDef.put(Material.DIORITE.name(), 0.6D);
+        materialSpeedsDef.put(Material.GRANITE.name(), 0.6D);
+        materialSpeedsDef.put(Material.ANDESITE.name(), 0.6D);
+        materialSpeedsDef.put(Material.NETHERRACK.name(), 0.6D);
+        if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_16)) {
+            materialSpeedsDef.put(Material.BLACKSTONE.name(), 0.6D);
+        }
+
+        addItemSetting(materialSpeeds);
+    }
+
+    @Override
+    public void preRegister() {
+        super.preRegister();
+
+        addItemHandler(getBlockBreakHandler());
+    }
+
+    
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            if (!e.getClickedBlock().isPresent()) return;
+
+            Block block = e.getClickedBlock().get();
+            ItemStack item = e.getItem();
+            Player p = e.getPlayer();
+
+            if (!getID().startsWith("TEST_CLIMBING_PICK") && p.getLocation().distance(block.getLocation()) > 1.5) return;
+            if (e.getClickedFace() == BlockFace.DOWN || e.getClickedFace() == BlockFace.UP) return;
+
+            if (!users.contains(p.getUniqueId())) {
+                Material mat = block.getType();
+                Double launch = materialSpeeds.getValue().get(mat.name());
+
+                Vector velocity = new Vector(0, 0, 0);
+                if (launch != null) {
+                    Integer efficiencyLevel = item.getEnchantments().get(Enchantment.DIG_SPEED);
+                    if (efficiencyLevel != null){
+                        launch += (efficiencyLevel * 0.2);
+                    }
+                    velocity.setY(launch);
+
+                    users.add(p.getUniqueId());
+                    Bukkit.getScheduler().runTaskLaterAsynchronously(SlimefunPlugin.instance, () -> users.remove(p.getUniqueId()), 4L);
```

Comment:
1. Why 4?
2. Why asynchronously? If you are accessing this Set asynchronously, you should absolutely use a thread-safe implementation like ConcurrentSet.

Reply:
I tested and 4 is the most ideal to actually climb. After 5-6 ticks the player starts to fall down.

Reply:
ah that needs to be sync...